### PR TITLE
Atmos tank console sensor fix.

### LINKED ISSUE #40858: Reconstructing N2O Supply Console makes for a BAD time
Round: 95274

Test merges:
#40747: 'Gives the curator EVA access' by vuonojenmustaturska at commit ac82a3f288
#40348: 'Fire/Lava Fixes & Tweaks' by ShizCalev at commit 8b5f4f7c42
#40821: 'Adjusts ignition snowflake in hotspot_expose to enable welders etc to start plasma fires again' by vuonojenmustaturska at commit 92f548efce
#40778: 'z_travel_pull_fix' by Dennok at commit 1d239c8cfa

N2O supply console got broken in an explosion, I replaced the glass and reconnected it to the N2O supply. It then began listing an ungodly number of gases and other stats, see gifv below.

[Imgur](https://i.imgur.com/0HvxwMX.gifv)

### PR DESCRIPTION
**Pull request by Dennok**
fixes #40858
:cl:
fix: Fix sensor set by reconnect button in atmos tank console.
add: Add rate button to atmos tank console to set injection rate.
/:cl:

### CODE REVIEW

**Comment by 81Denton** [comment]
Very nice. When the reconnect button works again, #41627 can be reverted since we will no longer need separate circuit boards for each tank.

File: code/game/machinery/computer/atmos_control.dm [hunk 1]
```diff
@@ -289,6 +290,12 @@ GLOBAL_LIST_EMPTY(atmos_air_controllers)
 		if("input")
 			signal.data += list("tag" = input_tag, "power_toggle" = TRUE)
 			. = TRUE
+		if("rate")
+			var/target = input("New target rate:", name, input_info ? input_info["volume_rate"] : 0) as num|null
+			if(!isnull(target) && !..())
+				target =  CLAMP(target, 0, 200)
```
Comment (yoyobatty) [inline code/game/machinery/computer/atmos_control.dm]: Shouldn't 200 be a define somewhere?

Comment (yoyobatty) [reply code/game/machinery/computer/atmos_control.dm]: [MAX_TRANSFER_RATE] I believe

Comment (Dennok) [reply code/game/machinery/computer/atmos_control.dm]: fixed

**Review by yoyobatty — APPROVED** [review]

File: code/game/machinery/computer/atmos_control.dm [hunk 2]
```diff
@@ -234,7 +234,7 @@ GLOBAL_LIST_EMPTY(atmos_air_controllers)
 // This hacky madness is the evidence of the fact that a lot of machines were never meant to be constructable, im so sorry you had to see this
```
Comment (Razharas) [inline code/game/machinery/computer/atmos_control.dm]: I cri eritiem

File: code/game/machinery/computer/atmos_control.dm [hunk 3]
```diff
@@ -289,6 +290,12 @@ GLOBAL_LIST_EMPTY(atmos_air_controllers)
 		if("input")
 			signal.data += list("tag" = input_tag, "power_toggle" = TRUE)
 			. = TRUE
+		if("rate")
+			var/target = input("New target rate:", name, input_info ? input_info["volume_rate"] : 0) as num|null
+			if(!isnull(target) && !..())
```
Comment (Cyberboss) [inline code/game/machinery/computer/atmos_control.dm]: Why are you calling ..() here?

Comment (Dennok) [reply code/game/machinery/computer/atmos_control.dm]: copypaste https://github.com/tgstation/tgstation/pull/41674/files/1e6ea0ffb6b19f03a1b6574a04a5ee706861d226#diff-20915160298ed00ab622c574f8f1c5a0R304

**Pull request merged**